### DISCUSSION TITLE
Why is the zero value of `ebiten.DrawImageOptions` null?

### DISCUSSION THREAD

**Discussion by musjj** [hajimehoshi/ebiten#2786] [Q&A]
```go
package main

import (
	"github.com/hajimehoshi/ebiten/v2"
)

type TestStruct struct {
	opts *ebiten.DrawImageOptions
	i    int
}

func main() {
	s := &TestStruct{}

	println(s.i) // 0
	println(s.opts) // 0x0
}
```

`s.i` is apparently a null value here, which I find *really* surprising. I thought the zero value of the struct would be initialized here?

On the other hand this works:

```go
s := &TestStruct{opts: &ebiten.DrawImageOptions{}}
println(s) // 0xc00019df00
```

Can anyone explain what is going on here? This footgun really terrifies me.

**Comment by musjj** [comment]
I think, I understand now. The zero value of a *pointer* is null.

**Reply by hajimehoshi** [reply]
That's right
